libp2p/protobuf/minprotobuf.nim

```diff
@@ -47,11 +51,18 @@ type
     of StartGroup, EndGroup:
       discard
 
-template protoHeader*(index: int, wire: ProtoFieldKind): uint =
+  ProtoResult {.pure.} = enum
+    VarintDecodeError,
+    MessageIncompleteError,
+    BufferOverflowError,
+    MessageSizeTooBigError,
+    NoError
+
+template getProtoHeader*(index: int64, wire: ProtoFieldKind): uint64 =
   ## Get protobuf's field header integer for ``index`` and ``wire``.
-  ((uint(index) shl 3) or cast[uint](wire))
+  ((uint64(index) shl 3) or cast[uint64](wire))
 
-template protoHeader*(field: ProtoField): uint =
+template getProtoHeader*(field: ProtoField): uint =
   ## Get protobuf's field header integer for ``field``.
   ((uint(field.index) shl 3) or cast[uint](field.kind))
```

Comment:
strictly, `field.index` should be `uint32` and range-checked when reading - https://developers.google.com/protocol-buffers/docs/proto3#assigning_field_numbers - ie 3 bits of header and 29 bits of filed numbers